src/main/java/com/google/devtools/build/lib/util/OptionsUtils.java

```diff
@@ -103,7 +104,11 @@ public static String asFilteredShellEscapedString(OptionsProvider options) {
 
     @Override
     public PathFragment convert(String input) {
-      return PathFragment.create(input);
+      String path = Preconditions.checkNotNull(input);
+      if (!path.isEmpty() && path.charAt(0) == '~') {
```

Comment:
Please add an `OS.getCurrent()` check around this `if`, and only replace "~" on Linux/macOS.

Reply:
(`OS` in the same Java package)

Reply:
AFAIK, `user.home` should be resolved just fine on Windows and recent JDK versions: [1]. On Windows, on Java 8, we have:

```java
C:\>type Test.java
public class Test {
  public static void main(String[] args) {
    String path = "~/foo/";
    path = path.replace("~", System.getProperty("user.home"));
    System.out.println(path);
  }
}
```

Running it produces:

```shell
C:\>java Test
C:\Users\davido/foo/
```

FTR: I backported this solution from that Buck feature: [Add support for user home in cache directory name](https://github.com/facebook/buck/commit/a1ba00118666c39f87354e3c0f2226e6f87c6970) that I added to Buck 4 years ago. And of course, Buck is platform independent.

[1] https://bugs.java.com/view_bug.do?bug_id=4787931
[2] https://stackoverflow.com/questions/2134338/java-user-home-is-being-set-to-userprofile-and-not-being-resolved

Reply:
Right, but `~abc` is a valid directory name on Windows. So I'm reluctant to expand "~" to user.home on Windows too.

Reply:
We need a working and platform inndependent solution to configure local action path per default and commit it in (D)VCS. I am open for any suggestions, but we just can't solve the probelm at hand for Linux/macOS, and ignore Windows users.

Should we compare for "~/" instead on all platforms, a lá:

````java
 if (foo.startWith("~/")) {
 [...]
```

Reply:
Yes please, that'd be better.

Reply:
Done.